src/components/modals/baseModal.vue

```diff
@@ -0,0 +1,30 @@
+<template>
+  <Teleport to="#modals">
+    <UseFocusTrap :options="options">
+      <div tabindex="0" class="fixed inset-0 bg-gray-900 bg-opacity-40">
+        <div
+          class="fixed inset-0 flex items-center justify-center w-auto m-auto max-w-[500px]"
+        >
+          <div class="p-4 bg-white text-black rounded-md w-full">
+            <slot></slot>
+          </div>
+        </div></div
+    ></UseFocusTrap>
+  </Teleport>
+</template>
+<script setup lang="ts">
+import { UseFocusTrap } from '@vueuse/integrations/useFocusTrap/component';
+import { UseFocusTrapOptions } from '@vueuse/integrations';
+const options: UseFocusTrapOptions = {
+  immediate: true,
+  onDeactivate: () => {
+    emit('dismiss', false);
+  },
+  tabbableOptions: {
+    displayCheck: 'none',
+  },
```

Comment:
Can you provide some details on why this option has been defined?

Reply:
I added this option to fix an error in my coverage test. The error message was the following:
Error: Your focus-trap must have at least one container with at least one tabbable node in it at all times.
activate node_modules/.pnpm/@vueuse+integrations@9.13.0_focus-trap@7.5.3_universal-cookie@4.0.4_vue@3.2.47/node_modules/@vueuse/integrations/useFocusTrap/component.mjs:11:41
      9|     let trap;
     10|     const target = ref();
     11|     const activate = () => trap && trap.activate();

Reply:
You should only activate this option in tests by mocking it in some ways. See https://github.com/focus-trap/tabbable#testing-in-jsdom

Reply:
I tried to mock the test but didn't work anyway.

Reply:
There are many ways to do this. A simple one could be:

```patch
diff --git a/src/components/modals/baseModal.vue b/src/components/modals/baseModal.vue
index 8ffb8d6..211023c 100644
--- a/src/components/modals/baseModal.vue
+++ b/src/components/modals/baseModal.vue
@@ -20,10 +20,16 @@ const options: UseFocusTrapOptions = {
   onDeactivate: () => {
     emit('dismiss', false);
   },
-  tabbableOptions: {
-    displayCheck: 'none',
-  },
 };
+
+// tests outside real browsers like jsdom require tabbableOptions.displayCheck
+// set to none (https://github.com/focus-trap/tabbable#testing-in-jsdom)
+if (import.meta.env.TEST) {
+  options.tabbableOptions = {
+    displayCheck: 'none',
+  };
+}
+
 const emit = defineEmits<{
   (e: 'dismiss', value: boolean): void;
 }>();
```

Reply:
@sgotti Thanks for the suggested solution. 
I also have another problem testing baseModal.test.ts that I cannot resolve.
If I remove the line 5 ' it('renders modal and handles focus trapping', async () => {' 
I get this error: TypeError: Cannot read properties of undefined (reading '0')
 ❯ src/components/modals/baseModal.test.ts:20:10
     18|   const dismissEvent = wrapper.emitted('dismiss');
     19|   // expect(dismissEvent).toBeTruthy();
     20|   expect(dismissEvent![0]).toEqual([false]);

I tried different solutions but without a happy end.
Do you have some suggestions?

Reply:
@raynasorint I'm not sure I understand what you're trying to do. Push a commit somewhere with the related changes.

Reply:
Create a new branch on your fork and push a commit with the changes that cause issues.

Reply:
Done!
https://github.com/raynasorint/agola-web/tree/ui--modal-infrastrucure-test-issue
Thank you

Reply:
the call to `it(...`  never executed the test since you wrapped it by an async function that's never awaited. So removing `it(...)` made the test run and it failed because it wasn't correct.

This is a working test

```js
import { mount } from '@vue/test-utils';
import baseModal from './baseModal.vue';

test('baseModal', async () => {
  const wrapper = mount(baseModal, {
    props: { show: true },
    global: {
      stubs: {
        teleport: true,
      },
    },
    attachTo: document.body,
  });
  // wait for focus trap to be ready
  await wrapper.vm.$nextTick();

  await wrapper.find('[tabindex="0"]').trigger('keydown', { key: 'Escape' });

  expect(wrapper.emitted().dismiss).toHaveLength(1);
  expect(wrapper.emitted().dismiss[0]).toEqual([false]);
});
```

The primary issue was that the component by default isn't attached to the document so focus trap wasn't receiving the key events.

Also note the use of `expect(wrapper...` like explained in the vitest docs to avoid the 'non-null assertion' that's reported as a warning by eslint. You should do the same also in the confirmationDialog tests.

Reply:
@sgotti thank you so much for the help.
I tried to add the nextTrick() function but I didn't attached the primary component so I think that was the problem.

I pit the correct code now.

Thanks